cobra/cmd/init.go

```diff
@@ -14,42 +14,41 @@
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+	"os/exec"
 	"path"
+	"path/filepath"
+	"strings"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 var (
-	pkgName string
-
 	initCmd = &cobra.Command{
-		Use:     "init [name]",
+		Use:     "init [path]",
 		Aliases: []string{"initialize", "initialise", "create"},
 		Short:   "Initialize a Cobra Application",
 		Long: `Initialize (cobra init) will create a new application, with a license
 and the appropriate structure for a Cobra-based CLI application.
 
-  * If a name is provided, a directory with that name will be created in the current directory;
-  * If no name is provided, the current directory will be assumed;
+Cobra init must be run inside of a go module (please run "go mod init <MODNAME>" first)
 `,
 
 		Run: func(_ *cobra.Command, args []string) {
-
 			projectPath, err := initializeProject(args)
 			cobra.CheckErr(err)
+			cobra.CheckErr(goGet("github.com/spf13/cobra"))
+			if viper.GetBool("useViper") {
+				cobra.CheckErr(goGet("github.com/spf13/viper"))
+			}
 			fmt.Printf("Your Cobra application is ready at\n%s\n", projectPath)
 		},
 	}
 )
 
-func init() {
-	initCmd.Flags().StringVar(&pkgName, "pkg-name", "", "fully qualified pkg name")
```

Comment:
Because `pkg-name` has been an available flag for a few releases now, i think we should deprecate it instead of removing it entirely.

Reply:
The problem is that it doesn't really make sense anymore with modules and can actually create a pretty broken experience for users who follow old documentation without knowing what it is doing (especially if the pkgname doesn't align with the module). Since this is a tool and not a dependency anyone depends on I'd rather let them know it's changed as early as possible and let them view the updated help.

Reply:
In other words, I can't think of a single reason (but there could be one I'm not thinking of) that leaving this option does any good. We can already get the right name from the module. The module needs to be created independently anyway. If we require a module to be initialized first then we can guarantee to use the right name here and keep them in sync without asking the user to do an extra step. 

I debated keeping this and then doing the `go mod init` step if no module is present and concluded that that was a terrible idea and rife with unexpected behaviors and edge cases. 

I think having the tool work simply and consistently is best for everyone.

Reply:
Ah I see. I'm okay with this approach :+1: